### PR TITLE
WIP: Add Python 3.12 to CI build matrix

### LINKED ISSUE #30: Check Py3.12 compatibility
I believe there are some issues with Py3.12 that I'm seeing upstream with the core library. We should check if things work here and port back any fixes.

### PR DESCRIPTION
**Pull request by ianmkenney**
Fixes #30 

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by IAlibay** [comment]
MDA doesn't support 3.12 yet

**Comment by ianmkenney** [comment]
> MDA doesn't support 3.12 yet

Seems so, would you like it if I keep this open for when MDA does or just close the PR and address it later?

**Comment by IAlibay** [comment]
I'm always of the view that PRs pending upstream changes should be kept open.

File: .github/workflows/gh-ci.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
         matrix:
           # os: [macOS-latest, ubuntu-latest, windows-latest]
           os: [ubuntu-latest, macOS-latest]
-          python-version: ["3.9", "3.10", "3.11"]
+          python-version: ["3.9", "3.10", "3.11", "3.12"]
```
Comment (IAlibay) [inline .github/workflows/gh-ci.yaml:46]: If you put an exclude for the combination of py3.12 + latest then this should work

**Comment by IAlibay** [comment]
Closing as superseded by #42

**Pull request closed without merging**